File: 040-049/ABC045_B.c
// ABC045 B - 3人でカードゲームイージー

#include <stdio.h>

int main(void) {
    char A[101];
    char B[101];
    char C[101];

    // input
    scanf("%s", A);
    scanf("%s", B);
    scanf("%s", C);

    int a_top = 0, b_top = 0, c_top = 0;
    char next;
    char winner;

    next = A[a_top++];

    while(1) {
        if (next == 'a') {
            next = A[a_top++];
            if (next =='\0') {
                winner = 'A';
                break;
            }
        } else if (next == 'b') {
            next = B[b_top++];
            if (next == '\0') {
                winner = 'B';
                break;
            }
        } else if (next == 'c') {
            next = C[c_top++];
            if (next == '\0') {
                winner = 'C';
                break;
            }
        }
    }

    // output
    printf("%c\n", winner);

    return 0;

}

File: 040-049/ABC045_B.js
'use strict';

function Main(input) {
  const tmp = input.split('\n');
  const A = tmp[0].split('').reverse();
  const B = tmp[1].split('').reverse();
  const C = tmp[2].split('').reverse();

  let winner;
  let turn = 1;
  let next = A.pop();
  while (turn) {
    switch (next) {
      case 'a':
        next = A.pop();
        if (next == undefined) {
          winner = 'A';
          turn = 0;
        }
        break;
      case 'b':
        next = B.pop();
        if (next == undefined) {
          winner = 'B';
          turn = 0;
        }
        break;
      case 'c':
        next = C.pop();
        if (next == undefined) {
          winner = 'C';
          turn = 0;
        }
        break;
    }
  }
  console.log(winner);
}
Main(require('fs').readFileSync('/dev/stdin', 'utf8'));
